include/zephyr/zbus/zbus.h

```diff
@@ -379,6 +387,37 @@ k_timeout_t _zbus_timeout_remainder(uint64_t end_ticks);
  */
 #define ZBUS_LISTENER_DEFINE(_name, _cb) ZBUS_LISTENER_DEFINE_WITH_ENABLE(_name, _cb, true)
 
+/**
+ * @brief Define and initialize a message subscriber.
+ *
+ * This macro defines an observer of message subscriber type. It defines a FIFO where the
+ * subscriber will receive the message asynchronously and initialize the ``struct
+ * zbus_observer`` defining the subscriber.
+ *
+ * @param[in] _name The subscriber's name.
+ * @param[in] _enable The subscriber's initial state.
+ */
+#define ZBUS_MSG_SUBSCRIBER_DEFINE_WITH_ENABLE(_name, _enable)                                     \
+	static K_FIFO_DEFINE(_zbus_observer_fifo_##_name);                                         \
+	STRUCT_SECTION_ITERABLE(zbus_observer, _name) = {                                          \
+		ZBUS_OBSERVER_NAME_INIT(_name) /* Name field */                                    \
+			.type = ZBUS_OBSERVER_MSG_SUBSCRIBER_TYPE,                                 \
+		.enabled = _enable,                                                                \
```

Comment:
I still think that the enable bit should go from this structure and discarding an event should be handled by the caller. Imagine having more parameters to set, how many variations of this would you end up having?

Reply:
I could not find a good solution for that. If you look at the observation structure, I could put the enabled flag outside. But in this case, we sometimes have to access the observer struct directly, and using external is not performant (or maybe feasible).

Reply:
Another important aspect is that you can even disable the whole observer by setting this flag false. Alternatively, you can mask a notification from a specific channel by selecting the individual observation as masked.